dagshub/common/cli.py

```diff
@@ -105,8 +114,77 @@ def upload(ctx,
 
     owner, repo_name = repo
     repo = Repo(owner=owner, name=repo_name, branch=branch)
-    repo.upload(file=filename, path=target, commit_message=message, force=update)
+    try:
+        repo.upload(file=filename, path=target, commit_message=message, force=update)
+    except RuntimeError as re:
+        # if uploaded filename already exists but update is False, a strange error message will appear
+        # since gogs tries to check last commit in this case
+        if "invalid last_commit" in str(re):
+            raise RuntimeError('Verify that the uploaded file is new') from re
+        else:
+            raise re
+
+
+@cli.command()
+@click.argument("repo", callback=validate_repo)
+@click.option("-u", "--upload-data", help="Upload data from specified url to new repository")
+@click.option("-c", "--clone", is_flag=True,  help="Clone repository locally")
+@click.option("-v", "--verbose", default=0, count=True, help="Verbosity level")
+def repo_create(
```

Comment:
Use [@cli.group()](https://click.palletsprojects.com/en/8.1.x/commands/#merging-multi-commands) so we can have `dagshub repo create`.
We might extend it in the future to be `dagshub repo delete` or something else

Reply:
**it can be done easily, but the help looks wiered when i do it:** 

`dagshub      
Usage: dagshub [OPTIONS] COMMAND [ARGS]...

Options:
  --host TEXT  Hostname of DagsHub instance
  --help       Show this message and exit.

Commands:
  login   Initiate an Oauth authentication process.
  mount   Mount a DagsHub Storage folder via FUSE
  repo    commands on a repo: create
  upload  Upload FILENAME to REPO at location TARGET.

(venv_test) yuvaldafni@h-MacBook-Pro-sl-yuval yuv_tutorial % dagshub repo
Usage: dagshub repo [OPTIONS] COMMAND [ARGS]...

  commands on a repo: create

Options:
  --help  Show this message and exit.

Commands:
  create  create a repo, example: dagshub repo create mytutorial -u...
`

so i prefer to do it only when we realy need more than one command

Reply:
You can do custom help with a docstring
```
@cli.group()
def repo():
    """Working with DagsHub repositories"""

```

Reply:
done (pending push)